### PR TITLE
Support external spire

### PR DESCRIPTION
**Pull request by danielpcox**
[sc-13415]

### CODE REVIEW

File: inputs.cue [hunk 1]
```diff
@@ -7,7 +7,8 @@ import (
 
 config: {
 	// Flags
-	spire:                       bool | *false           @tag(spire,type=bool)           // enable Spire-based mTLS
+	spire:                       bool | *false           @tag(spire,type=bool)           // use Spire-based mTLS (ours or another)
+	deploy_spire:                bool | *spire           @tag(use_spire,type=bool)       // deploy our own server and agent
```
Comment (dontlaugh) [inline inputs.cue:11]: Let's move these comments above the fields they're commenting on

**Review by dontlaugh — CHANGES_REQUESTED** [review]
Just one nitpick, the rest seems fine

File: inputs.cue (hunk 1 above)
Comment (danielpcox) [reply inputs.cue:11]: Done, and added some extras

**Review by dontlaugh — COMMENTED** [review]
Rebase and squash this to 1) remove conflicts with main and 2) clean up the commit messages

**Comment by danielpcox** [comment]
This is a branch off `multiple-operators-in-one-cluster`, so I'll do that when that one's merged.

**Review by dontlaugh — COMMENTED** [review]
Just to confirm my understanding, `deploy_spire` being true means we deploy a spire server and agents (DaemonSet that mounts a host socket).

I think there is another configuration - deploy agents as a DaemonSet, but configure them to point to an external server. Is that possible here?

**Comment by danielpcox** [comment]
The other configuration I know about deploys the spire agents as injected containers into the same pod, and each of our sidecars communicates with its local agent. I tried to make that possible with CUE-only changes here.

**Comment by danielpcox** [comment]
And yeah, `deploy_spire` being true means we deploy our own spire server (StatefulSet) and agent(s) (DaemonSet).

**Review by dontlaugh — APPROVED** [review]
I won't block this PR on providing a more detailed SPIRE configuration.

However, I am concerned about the proliferation of booleans. They might invite unwarranted complexity. Let's think twice before introducing another.

**Comment by danielpcox** [comment]
Yeah. I'm right there with you. `inputs.cue` needs a more thorough cleanup in general, and that's one of the things to think about. Its original organization has become a bit obsolete now that we're doing GitOps.

**Review by dontlaugh — APPROVED** [review]

**Pull request merged**